crypto/evp/mac_lib.c

```diff
@@ -48,18 +48,19 @@ void EVP_MAC_CTX_free(EVP_MAC_CTX *ctx)
     OPENSSL_free(ctx);
 }
 
-int EVP_MAC_CTX_copy(EVP_MAC_CTX *dst, EVP_MAC_CTX *src)
+EVP_MAC_CTX *EVP_MAC_CTX_dup(const EVP_MAC_CTX *src)
 {
-    EVP_MAC_IMPL *macdata;
+    EVP_MAC_CTX *dst = EVP_MAC_CTX_new(src->meth);
 
-    if (src->data != NULL && !dst->meth->copy(dst->data, src->data))
-        return 0;
+    if (dst == NULL)
+        return NULL;
 
-    macdata = dst->data;
-    *dst = *src;
-    dst->data = macdata;
+    if (src->data != NULL && src->meth->copy(dst->data, src->data) == 0) {
```

Comment:
If you want to do this, don't do it half way, change the method function too.

Reply:
I'm not sure what you mean. I have an other commit that changes the copy function to a dup function, it's just not pushed yet.

Reply:
Oh good, that is exactly what I was asking for

Reply:
Just pushed it

Reply:
Uhmmmm, this is still just half way?

I do see one instance where you use the copy method function directly, will comment there.